spec/lib/active_storage/service/ds_proxy_service.rb

```diff
@@ -0,0 +1,27 @@
+describe ActiveStorage::Service::DsProxyService do
+  let(:storage_service) { storage_service = double(ActiveStorage::Service) }
+  let(:private_url) { 'https://storage.sbg1.cloud.ovh.net:443/v1/AUTH_a24c37ed11a84896914514384898c34b/test_local/2R6rr89nFeSRkSgXHd3smvEf?temp_url_sig=5ab8cfc3ba5da2598a6c88cc6b1b461fe4e115bc&temp_url_expires=1547598179' }
+  let(:public_url) { 'https://static.demarches-simplifiees/test_local/2R6rr89nFeSRkSgXHd3smvEf?temp_url_sig=5ab8cfc3ba5da2598a6c88cc6b1b461fe4e115bc&temp_url_expires=1547598179' }
+
+  subject { ActiveStorage::Service::DsProxyService.new(wrapped: storage_service) }
+
+  describe '#url' do
+    before do
+      expect(storage_service).to receive(:url).and_return(private_url)
+    end
+
+    it do
```

Comment:
Pinaillage : j'aime bien quand les specs, well, spécifient – enfin quand elles écrivent en bon ~~français~~ anglais le comportement attendu.

Est-ce qu'un `it "rewrites the host of the storage url"` t'irait ?

Reply:
J’ai pas exactement mis ça, car ça passe sous silence la suppression de la partie AUTH_xyz, mais j’ai rajouté un texte, est-ce qu’il te convient ? Et j’en ai profité pour décomposer private_url et public_url pour que le lecteur ait une chance de voir ce qui bouge entre les deux.